File: contests/codechef/march12_cookoff/ceil.py
t = int(raw_input())
_3=0
_5=0
_8=0
ans=0

while t>0:
	t-=1
	s = raw_input().split()
	l = len(s)
	for ch in s[l-1]:
		if ch == '3': _3+=1
		elif ch == '5': _5+=1
		elif ch == '8': _8+=1
		else: _8=-1; break
	if (_8 >= _5 and _5 >= _3): ans+=1
print ans

File: contests/codechef/march12_cookoff/ceil.cpp
#include <cstdio>
#include <iostream>
#include <cstring>
using namespace std;

int main()
{
	int t; scanf("%d\n",&t);
	char s[1000000]; long long ans=0;
	while(t--)
	{
		gets(s);
		unsigned long len = strlen(s), _3,_5,_8; bool pos=1;
		_3 = _5 = _8 = 0;
		for(int a=0;a<len;a++)
		{
			if(!isdigit(s[a])) { _3 = _5 = _8 = 0; pos=1; continue; }
			if(s[a] == '3') _3++;
			else if(s[a] == '5') _5++;
			else if(s[a] == '8') _8++;
			else { pos=0; continue; }
		}
		if(_8 >= _5 && _5 >= _3 && pos) ans++;
	}
	printf("%lld\n",ans);
	return 0;
}
